### PR TITLE
Add TTL 74166

### PR DESCRIPTION
**Pull request by btb**
split off from #1461

### CODE REVIEW

**Review by maehne — DISMISSED** [review]
@btb: Thanks, looks good to me!

File: src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java:24]: Please, add a docstring describing the purpose of the class.

File: src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java:98]: Please remove.

File: src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java:101]: Please add `{}` to code blocks

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java
Comment (maehne) [inline src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java:25]: Please, be a bit more specific: "clear" -> "async clear"

File: src/main/resources/resources/logisim/strings/std/std.properties
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std.properties:777]: "clear" -> "async clear"

File: src/main/resources/resources/logisim/strings/std/std_nl.properties
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std_nl.properties:777]: "clear" -> "asynchrone clear"

File: src/main/resources/resources/logisim/strings/std/std_pl.properties
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std_pl.properties:777]: If possible, please be more precise to state that the clear is asynchronous.

@MarcinOrlowski: Could you maybe help?

File: src/main/resources/resources/logisim/strings/std/std_ru.properties
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std_ru.properties:777]: If possible, please be more precise to state that the clear is asynchronous.

File: src/main/resources/resources/logisim/strings/std/std_pt.properties
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std_pt.properties:777]: If possible, please be more precise to state that the clear is asynchronous.

File: src/main/resources/resources/logisim/strings/std/std_de.properties
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std_de.properties:776]: "Parallel zu Serielles" -> "parallel-zu-seriell"

File: src/main/resources/resources/logisim/strings/std/std_de.properties
Comment (maehne) [inline src/main/resources/resources/logisim/strings/std/std_de.properties:777]: "Parallel zu Serielles" -> "parallel-zu-seriell"
"Clear" -> "asynchronem Clear"

**Review by maehne — CHANGES_REQUESTED** [review]
Thanks for addressing my comments! Looks mostly good to me! Though, it would be nice if you could address @MarcinOrlowski's and mine additional suggestions.

File: src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java
Comment (btb) [reply src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java]: done

File: src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java
Comment (btb) [reply src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java]: done

File: src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java
Comment (btb) [reply src/main/java/com/cburch/logisim/std/ttl/Ttl74166.java]: done

File: src/main/resources/resources/logisim/strings/std/std_nl.properties
Comment (btb) [reply src/main/resources/resources/logisim/strings/std/std_nl.properties]: done

File: src/main/resources/resources/logisim/strings/std/std_pl.properties
Comment (btb) [reply src/main/resources/resources/logisim/strings/std/std_pl.properties]: done

File: src/main/resources/resources/logisim/strings/std/std_ru.properties
Comment (btb) [reply src/main/resources/resources/logisim/strings/std/std_ru.properties]: done

File: src/main/resources/resources/logisim/strings/std/std_de.properties
Comment (btb) [reply src/main/resources/resources/logisim/strings/std/std_de.properties:776]: will fix in upcoming PR to fix #1472

File: src/main/resources/resources/logisim/strings/std/std_de.properties
Comment (btb) [reply src/main/resources/resources/logisim/strings/std/std_de.properties]: done

File: src/main/resources/resources/logisim/strings/std/std_es.properties
Comment (MarcinOrlowski) [inline src/main/resources/resources/logisim/strings/std/std_es.properties:777]: Please remove anything but the key. So it should be just

```ini
# ==> TTL74166
```

applies to all the cases below as well. One can always use [trans-tool](https://github.com/MarcinOrlowski/trans-tool) to get reference strings during works on their translation

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/resources/resources/logisim/strings/std/std_es.properties
Comment (btb) [reply src/main/resources/resources/logisim/strings/std/std_es.properties:777]: existing empty translations all include the equal sign as well, i.e.
```
# ==> TTL74166 =
```

should it not be done that way?

Comment (MarcinOrlowski) [reply src/main/resources/resources/logisim/strings/std/std_es.properties:777]: It's not the case for all the translation files, hence my request.

File: src/main/resources/resources/logisim/strings/std/std_pl.properties
Comment (MarcinOrlowski) [inline src/main/resources/resources/logisim/strings/std/std_pl.properties:777]:
```ini
TTL74166 = 74166: 8-bitowy rejestr przesunięcia równoległego do szeregowego z kasowaniem asynchronicznym
```

File: src/main/resources/resources/logisim/strings/std/std_pt.properties
Comment (MarcinOrlowski) [inline src/main/resources/resources/logisim/strings/std/std_pt.properties:777]: Please leave just a key.

```ini
# ==> TTL74166 =
```

File: src/main/resources/resources/logisim/strings/std/std_ru.properties
Comment (MarcinOrlowski) [inline src/main/resources/resources/logisim/strings/std/std_ru.properties:777]: Please leave just a key.

```ini
# ==> TTL74166 =
```

File: src/main/resources/resources/logisim/strings/std/std_nl.properties
Comment (MarcinOrlowski) [inline src/main/resources/resources/logisim/strings/std/std_nl.properties:777]: Please leave just a key.

```ini
# ==> TTL74166 =
```

File: src/main/resources/resources/logisim/strings/std/std_it.properties
Comment (MarcinOrlowski) [inline src/main/resources/resources/logisim/strings/std/std_it.properties:777]: Please leave just a key.

```ini
# ==> TTL74166 =
```

File: src/main/resources/resources/logisim/strings/std/std_fr.properties
Comment (MarcinOrlowski) [inline src/main/resources/resources/logisim/strings/std/std_fr.properties:777]: Please leave just a key.

```ini
# ==> TTL74166 =
```

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

**Comment by maehne** [comment]
@MarcinOrlowski: What still needs to be done before we can merge this PR?

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski : What is still blocking this PR?

**Review by MarcinOrlowski — APPROVED** [review]

**Comment by MarcinOrlowski** [comment]
Sorry for the lag. Code looks OK for me.

**Pull request merged**
